src/main/java/net/glowstone/constants/GlowStatistic.java

```diff
@@ -127,12 +54,111 @@ private GlowStatistic() {
      * @return the statistic name.
      */
     public static String getName(Statistic stat) {
-        checkNotNull(stat, "Achievement cannot be null");
-        return names[stat.ordinal()];
+        checkNotNull(stat, "Statistic cannot be null");
+        return STATISTICS.get(stat);
+    }
+
+    /**
+     * Get the statistic name for a specified Statistic and Material.
+     *
+     * @param stat the Statistic
+     * @param material the Material
+     * @return the statistic name
+     */
+    public static String getName(Statistic stat, Material material) {
+        checkNotNull(stat, "Statistic cannot be null");
+        checkNotNull(material, "Material cannot be null");
+
+        if (SUB_STATISTICS.containsKey(stat)) {
+            return SUB_STATISTICS.get(stat).get(material);
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the statistic name for a specified Statistic and EntityType.
+     *
+     * @param stat the Statistic
+     * @param entityType the EntityType
+     * @return the statistic name
+     */
+    public static String getName(Statistic stat, EntityType entityType) {
+        checkNotNull(stat, "Statistic cannot be null");
+        checkNotNull(entityType, "EntityType cannot be null");
+
+        if (SUB_STATISTICS.containsKey(stat)) {
+            return SUB_STATISTICS.get(stat).get(entityType);
+        }
+
+        return null;
     }
 
-    private static void set(Statistic stat, String key) {
-        names[stat.ordinal()] = "stat." + key;
+    private static void set(Statistic statistic, Enum data, String key) {
+        if (!SUB_STATISTICS.containsKey(statistic)) {
+            SUB_STATISTICS.put(statistic, new HashMap<>());
+        }
+        SUB_STATISTICS.get(statistic).put(data, "stat." + key);
     }
 
+
+    private static void generateStatistics() {
+        for (Statistic stat : values()) {
+            STATISTICS.put(stat, "stat." + UPPER_UNDERSCORE.to(LOWER_CAMEL, stat.name()));
+        }
+
+        // Specific case
+        STATISTICS.put(PLAY_ONE_TICK, "stat.playOneMinute");
+    }
+
+    private static void generateMaterialStatistics() {
+        for (Material material : Material.values()) {
+            String name = material.name().toLowerCase()
```

Comment:
I think this should make use of the `ItemIds` class, which has a registry for such IDs.

Reply:
Good call, I missed this class. Thanks